spring-boot-samples/spring-boot-sample-data-geode/src/main/java/sample/data/geode/SampleDataGeodeApplication.java

```diff
@@ -0,0 +1,161 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sample.data.geode;
+
+import java.util.Properties;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.data.gemfire.CacheFactoryBean;
+import org.springframework.data.gemfire.GemfireTemplate;
+import org.springframework.data.gemfire.PartitionedRegionFactoryBean;
+import org.springframework.data.gemfire.RegionAttributesFactoryBean;
+
+import com.gemstone.gemfire.cache.Cache;
+import com.gemstone.gemfire.cache.CacheLoader;
+import com.gemstone.gemfire.cache.CacheLoaderException;
+import com.gemstone.gemfire.cache.LoaderHelper;
+import com.gemstone.gemfire.cache.RegionAttributes;
+
+import sample.data.geode.service.Calculator;
+import sample.data.geode.service.factory.CalculatorFactory;
+
+/**
+ * SampleDataGeodeApplication class is a Spring Boot, Apache Geode peer cache application using Geode
+ * to perform and store mathematical calculations.
+ *
+ * @author John Blum
+ * @see org.springframework.boot.SpringApplication
+ * @see org.springframework.boot.autoconfigure.SpringBootApplication
+ * @see org.springframework.boot.context.properties.ConfigurationProperties
+ * @see org.springframework.context.ApplicationContext
+ * @see org.springframework.context.annotation.Bean
+ * @see org.springframework.data.gemfire.CacheFactoryBean
+ * @see org.springframework.data.gemfire.GemfireTemplate
+ * @see org.springframework.data.gemfire.PartitionedRegionFactoryBean
+ * @see com.gemstone.gemfire.cache.Cache
+ * @see com.gemstone.gemfire.cache.CacheLoader
+ * @see sample.data.geode.service.Calculator
+ * @see sample.data.geode.service.factory.CalculatorFactory
+ * @since 1.4.0
+ */
+@SpringBootApplication
+@ConfigurationProperties
+@SuppressWarnings("unused")
+public class SampleDataGeodeApplication {
+
+	protected static final String CALCULATIONS_REGION_NAME = "Calculations";
+
+	public static void main(final String[] args) {
+		SpringApplication.run(SampleDataGeodeApplication.class, args);
+	}
+
+	@Bean
+	Properties gemfireProperties(@Value("${sample.data.geode.log.level:config}") String logLevel) {
```

Comment:
I am not entirely sure I understand the meaning of your comment...

1. I did declare the `@ConfigurationProperties` annotation on my `SampleDataGeodeApplication` class, which is the source for these "custom" sample application properties when explicitly specified by the user on the command-line while running the sample (manually, or with the test), defaulting otherwise.

2. However, they do allow a user to do the following when running the sample from the command-line (not/or within the IDE)...

`$ mvn -pl spring-boot-sample-data-geode clean test -Dsample.data.geode.log.level:config=warn`

I need help understanding the intent as well as the proper way to configure what you are stating.  I also assume (hope) the "custom key assistance in the IDE" is IDE agnostic?  I did not specifically think IDE when I coded my sample as I often do things from the command-line (and only occasionally in my IDE (IntelliJ IDEA)).